packages/ckeditor5-image/src/image/converters.ts

```diff
@@ -249,8 +242,24 @@ export function downcastSourcesAttribute( imageUtils: ImageUtils ): ( dispatcher
 				viewElement = parentElement;
 			}
 
-			// Insert the picture and move img into it.
-			viewWriter.insert( viewWriter.createPositionBefore( imgElement ), pictureElement );
+			const hasPictureElement = imgElement.parent!.is( 'element', 'picture' );
+
+			// Reuse existing <picture> element (ckeditor5#17192) or create a new one.
+			const pictureElement = hasPictureElement ? imgElement.parent : viewWriter.createContainerElement( 'picture', null );
+
+			if ( !hasPictureElement ) {
+				viewWriter.insert( viewWriter.createPositionBefore( imgElement ), pictureElement );
+			}
+
+			viewWriter.remove( viewWriter.createRangeIn( pictureElement ) );
+
+			// Make sure <picture> does not break attribute elements, for instance <a> in linked images.
```

Comment:
This comment looks unrelated to the code. It's a more generic comment about collecting attribute elements and reapplying them.

Reply:
True, I moved it with the code but it doesn't seem to fit there. Fixed 👍